src/phoenix/server/api/dataloaders/dataset_example_revisions.py

```diff
@@ -0,0 +1,101 @@
+from typing import (
+    AsyncContextManager,
+    Callable,
+    List,
+    Optional,
+    Tuple,
+)
+
+from sqlalchemy import Integer, case, func, literal, or_, select, union_all
+from sqlalchemy.ext.asyncio import AsyncSession
+from strawberry.dataloader import DataLoader
+from typing_extensions import TypeAlias
+
+from phoenix.db import models
+from phoenix.server.api.types.DatasetExampleRevision import DatasetExampleRevision
+
+ExampleID: TypeAlias = int
+VersionID: TypeAlias = Optional[int]
+Key: TypeAlias = Tuple[ExampleID, Optional[VersionID]]
+Result: TypeAlias = DatasetExampleRevision
+
+
+class DatasetExampleRevisionsDataLoader(DataLoader[Key, Result]):
+    def __init__(self, db: Callable[[], AsyncContextManager[AsyncSession]]) -> None:
+        super().__init__(load_fn=self._load_fn)
+        self._db = db
+
+    async def _load_fn(self, keys: List[Key]) -> List[Result]:
+        # sqlalchemy has limited SQLite support for VALUES, so use UNION ALL instead.
+        # For details, see https://github.com/sqlalchemy/sqlalchemy/issues/7228
+        keys_subquery = union_all(
```

Comment:
Can we use union to remove duplicates just in case? (Maybe there can't be any duplicates.)
```suggestion
        keys_subquery = union(
```

Reply:
Addressed in 3578757436f75e7ea5213201d6dbe326e4996360